### DISCUSSION TITLE
Expanding existing distributed not blances data well

### DISCUSSION THREAD

**Discussion by feythin** [minio/minio#14904] [Q&A]
I have a 4 node minio cluster，and disk used 75%。so I follow this guide （https://docs.min.io/docs/distributed-minio-quickstart-guide.html） to expanding distributed ，and it works well。but the new put object also store data in the origin cluster，which cause the origin cluster disk almost full。
$ minio --version
minio version RELEASE.2021-02-01T22-56-52Z

expanding distributed 
minio server --config-dir /data/opt/minio/conf http://10.1.1.{10...13}/data/minio{1...2} http://10.1.1.{14...17}/data/minio{1...4}
the origin cluster disk usage（10.1.1.{10...13}/data/minio1）
![image](https://user-images.githubusercontent.com/1758752/167982630-fac88511-604c-4493-befe-65b4dc48b9d5.png)
the expanding distributed cluster disk usage (10.1.1.{14...17}/data/minio1)
![image](https://user-images.githubusercontent.com/1758752/167982844-7cdbd228-65bf-48a1-8f9c-eda63f157bc7.png)

**Comment by klauspost** [comment]
Without any numbers on your graph this is hard to give any advise on. It is just two graphs going up over time.

If your "old" cluster has 75% used space, assuming the second cluster is of the same size, it will see ~75% of all writes land there.

**Reply by feythin** [reply]
klauspos，thanks for your reply。
at the begining of expanding  cluster。the "old" cluster has 75% used space， and the second cluster has used 0% 。but now the "old" cluster has 88% used space，and the second cluster has used 35%。 when I expanding cluster, It is should be all writes  land the second cluster until the size of used is the same as the "old" cluster, but it is not.

**Reply by klauspost** [reply]
> It is should be all writes land the second cluster until the size of used is the same as the "old" cluster, but it is not

The writes are distributed based on free space, so it will become less and less to the new. Also, if you are writing new versions of objects they are going to the same cluster as the original object.

**Reply by feythin** [reply]
I worried about the usage of the "old" cluster and storage backend has reached its minimum free disk threshold。

**Reply by harshavardhana** [reply]
> I worried about the usage of the "old" cluster and storage backend has reached its minimum free disk threshold。

You don't have to, MinIO will try to make sure that all servers are eventually fully utilized. 

If you are worried about production support you are advised to pay for support https://min.io/pricing so that you don't have to squander around in community threads for help.
